src/parameters.py

```diff
@@ -9,7 +9,8 @@ class IOParameters(BaseModel):
     mask_tiled_api_key: Optional[str] = Field(default=None, description="tiled api key for masks")
     seg_tiled_uri: Optional[str] = Field(default=None, description="tiled uri for segmenation results")
     seg_tiled_api_key: Optional[str] = Field(default=None, description="tiled api key for segmentation results")
-    uid: str = Field(description="uid for the workflow")
+    uid_save: str = Field(description="uid to save models, metrics and etc")
+    uid_retrieve: str = Field(description="uid to retrieve models")
```

Comment:
Correct, `uid_retrieve` is not used in training, the information I got from yesterday's discussion was: we will always get 2 uids, in training the `uid_retrieve` will be set to the same as `uid_retrieve` while in inference these will be specified individually.

I will go ahead and change that to Optional. Done.

Reply:
Following up: since I will only get 1 yaml file when we hit the `train` button, and 2 subflows will use the same yaml file (I assume), will `uid_retrieve` always be there?

Reply:
You bring up a really good point. The parent flow that runs training + partial inference will parse 1 `uid` (the parent flow id) to all the sub flows, as `uid_save`, so the partial inference will not receive a `uid_retrieve`. This is not a problem for full inference since `uid_retrieve` is populated within the Dash app. I can think about 2 ways of fixing this:

1. The parent flow in MLExWorker parses the `flow_id` to both `uid_save` and `uid_retrieve` for consecutive flows.
2. The DLSIA image in this repo uses `uid_save` by default if `uid_retrieve=None`

To be honest, I am not a huge fan of either approach. Ideally, we may want to use the sub flows ids to fill in each `uid_save` and `uid_retrieve` accordingly, but this would make retrieving the results back to the app more complicated.

Any thoughts? @Wiebke @dylanmcreynolds

Reply:
With the most recent PR in prefect_worker addressed this issue (good work!), I will change back the `uid_retrieve` to be not Optional.

Reply:
This has been addressed in mlexchange/mlex_prefect_worker#16 and no further changes are needed in dlsia

Reply:
Addressed based on this morning's discussion. `uid_retrieve` stays as Optional.